app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitTrackChairScoreTypesApiController.php

```diff
@@ -0,0 +1,257 @@
+
+        $selection_plan = $this->selection_plan_repository->getById(intval($selection_plan_id));
+        if (is_null($selection_plan)) return $this->error404();
+
+        $track_chair_rating_type = $selection_plan->getTrackChairRatingTypeById(intval($type_id));
+        if (is_null($track_chair_rating_type)) return $this->error404();
+
+        return $this->_getAll(
+            function () {
+                return [
+                    'summit_id' => ['=='],
+                    'selection_plan_id' => ['=='],
+                    'type_id' => ['=='],
+                ];
+            },
+            function () {
+                return [
+                    'summit_id' => 'sometimes|integer',
+                    'selection_plan_id' => 'sometimes|integer',
+                    'type_id' => 'sometimes|integer',
+                ];
+            },
+            function () {
+                return [];
+            },
+            function ($filter) use ($summit, $selection_plan, $track_chair_rating_type) {
+                if ($filter instanceof Filter) {
+                    $filter->addFilterCondition(FilterElement::makeEqual('summit_id', $summit->getId()));
+                    $filter->addFilterCondition(FilterElement::makeEqual('selection_plan_id', $selection_plan->getId()));
+                    $filter->addFilterCondition(FilterElement::makeEqual('type_id', $track_chair_rating_type->getId()));
+                }
+                return $filter;
+            },
+            function () {
+                $current_user = $this->resource_server_context->getCurrentUser();
+                if(!is_null($current_user)){
+                    if(
+                        $current_user->isOnGroup(IGroup::Administrators) ||
+                        $current_user->isOnGroup(IGroup::SuperAdmins) ||
+                        $current_user->isOnGroup(IGroup::TrackChairsAdmins) ||
+                        $current_user->isOnGroup(IGroup::SummitAdministrators)
+                    )
+                        return SerializerRegistry::SerializerType_Private;
+                }
+                return SerializerRegistry::SerializerType_Public;
+            }
+        );
+    }
+
+    /**
+     * @param $summit_id
+     * @param $selection_plan_id
+     * @param $type_id Track Chair Rating Type Id
+     * @param $score_type_id
+     * @return \Illuminate\Http\JsonResponse|mixed
+     */
+    public function getTrackChairScoreType($summit_id, $selection_plan_id, $type_id, $score_type_id) {
+
+        return $this->processRequest(function () use ($summit_id, $selection_plan_id, $type_id, $score_type_id) {
+            $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->resource_server_context)->find($summit_id);
+            if (is_null($summit)) return $this->error404();
+
+            $selection_plan = $this->selection_plan_repository->getById(intval($selection_plan_id));
```

Comment:
u need to do intval for all params that are supossed to be int

Reply:
done